internal/services/streamanalytics/stream_analytics_function_javascript_udf_resource.go

```diff
@@ -71,6 +71,12 @@ func resourceStreamAnalyticsFunctionUDF() *pluginsdk.Resource {
 								"record",
 							}, false),
 						},
+
+						"is_configuration_parameter": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
```

Comment:
I'm wondering whether this might be better as it's own block
```suggestion

			"configuration_parameter": {
				Type:     pluginsdk.TypeList,
				Required: true,
				MinItems: 1,
				Elem: &pluginsdk.Resource{
					Schema: map[string]*pluginsdk.Schema{
						"type": {
							Type:     pluginsdk.TypeString,
							Required: true,
							ValidateFunc: validation.StringInSlice([]string{
								"any",
								"datetime",
								"array",
								"bigint",
								"float",
								"nvarchar(max)",
								"record",
							}, false),
```

Reply:
@stephybun , In the future, if service team expands the sub properties of the root property "FunctionInput", then we have to maintain two blocks. So I assume it's better to align with service api. Does it make sense?

Reply:
Is there any different between a configuration and non configuration input parameter?

Reply:
@katbyte , 

For now, there is no different between them. Below is potential schema.
```
"input": {
    Type:     pluginsdk.TypeList,
    Required: true,
    MinItems: 1,
    Elem: &pluginsdk.Resource{
        Schema: map[string]*pluginsdk.Schema{
            "type": {
                Type:     pluginsdk.TypeString,
                Required: true,
                ValidateFunc: validation.StringInSlice([]string{
                    "any",
                    "datetime",
                    "array",
                    "bigint",
                    "float",
                    "nvarchar(max)",
                    "record",
                }, false),
            },
        },
    },
},

"configuration_parameter": {
    Type:     pluginsdk.TypeList,
    Required: true,
    MinItems: 1,
    Elem: &pluginsdk.Resource{
        Schema: map[string]*pluginsdk.Schema{
            "type": {
                Type:     pluginsdk.TypeString,
                Required: true,
                ValidateFunc: validation.StringInSlice([]string{
                    "any",
                    "datetime",
                    "array",
                    "bigint",
                    "float",
                    "nvarchar(max)",
                    "record",
                }, false),
            },
        },
    },
},
```

In the future, if service team expands the sub properties of the root property "FunctionInput", then we have to maintain two blocks and append these same sub properties in these two blocks. Below is the potential schema.
```
"input": {
    Type:     pluginsdk.TypeList,
    Required: true,
    MinItems: 1,
    Elem: &pluginsdk.Resource{
        Schema: map[string]*pluginsdk.Schema{
            "type": {
                Type:     pluginsdk.TypeString,
                Required: true,
                ValidateFunc: validation.StringInSlice([]string{
                    "any",
                    "datetime",
                    "array",
                    "bigint",
                    "float",
                    "nvarchar(max)",
                    "record",
                }, false),
            },

            "A": {
                ...
            },

            "B": {
                ...
            },

            ...
        },
    },
},

"configuration_parameter": {
    Type:     pluginsdk.TypeList,
    Required: true,
    MinItems: 1,
    Elem: &pluginsdk.Resource{
        Schema: map[string]*pluginsdk.Schema{
            "type": {
                Type:     pluginsdk.TypeString,
                Required: true,
                ValidateFunc: validation.StringInSlice([]string{
                    "any",
                    "datetime",
                    "array",
                    "bigint",
                    "float",
                    "nvarchar(max)",
                    "record",
                }, false),
            },

            "A": {
                ...
            },

            "B": {
                ...
            },

            ...
        },
    },
},
```

And if service team expands new types of FunctionInput via introducing new sub properties, then we have to maintain more blocks for them.
```
"input": {
    Type:     pluginsdk.TypeList,
    Required: true,
    MinItems: 1,
    Elem: &pluginsdk.Resource{
        Schema: map[string]*pluginsdk.Schema{
            "type": {
                Type:     pluginsdk.TypeString,
                Required: true,
                ValidateFunc: validation.StringInSlice([]string{
                    "any",
                    "datetime",
                    "array",
                    "bigint",
                    "float",
                    "nvarchar(max)",
                    "record",
                }, false),
            },

            "A": {
                ...
            },

            ...
        },
    },
},

"configuration_parameter": {
    Type:     pluginsdk.TypeList,
    Required: true,
    MinItems: 1,
    Elem: &pluginsdk.Resource{
        Schema: map[string]*pluginsdk.Schema{
            "type": {
                Type:     pluginsdk.TypeString,
                Required: true,
                ValidateFunc: validation.StringInSlice([]string{
                    "any",
                    "datetime",
                    "array",
                    "bigint",
                    "float",
                    "nvarchar(max)",
                    "record",
                }, false),
            },

            "A": {
                ...
            },

            ...
        },
    },
},

"D_Other_Input_paramter_type": {
    Type:     pluginsdk.TypeList,
    Required: true,
    MinItems: 1,
    Elem: &pluginsdk.Resource{
        Schema: map[string]*pluginsdk.Schema{
            "type": {
                Type:     pluginsdk.TypeString,
                Required: true,
                ValidateFunc: validation.StringInSlice([]string{
                    "any",
                    "datetime",
                    "array",
                    "bigint",
                    "float",
                    "nvarchar(max)",
                    "record",
                }, false),
            },

            "A": {
                ...
            },

            ...
        },
    },
},

...
```

So I assume it would increase the cost of maintanence. Hence, I assume it's better to align with service api. Does it make sense? If I am wrong, please correct me. Thanks.

Reply:
this makes sense, however it is quite an odd name. A single block sounds good but could we change the property name to `configuration_parameter`, making an exception for bools as is really stands out as wrong to me. WDYT?

Reply:
Updated. Please take another look. Thanks.